toqito/matrices/operator_sinkhorn.py

```diff
@@ -0,0 +1,195 @@
+"""Sinkhorn operator."""
+
+import warnings
+
+import numpy as np
+import scipy as sp
+
+from toqito.channels.partial_trace import partial_trace
+from toqito.matrix_props.is_square import is_square
+
+
+def operator_sinkhorn(
+    rho: np.ndarray, dim: list[int] = None, tol: float = np.sqrt(np.finfo(float).eps), max_iterations: int = 10000
+) -> tuple[np.ndarray, list[np.ndarray]]:
+    r"""Perform the operator Sinkhorn iteration.
+
+    This function implements the iterative Sinkhorn algorithm to find a density matrix
+    that is locally equivalent to a given bipartite or multipartite density matrix
+    `rho`, but has both of its partial traces proportional to the identity.
+
+    Examples
+    ==========
+
+    To Perform Operator Sinkhorn on a random 2-Qubit Bi-partite state:
+
+    >>> import numpy as np
+    >>> from toqito.matrices import operator_sinkhorn
+    >>> from toqito.rand import random_density_matrix
+
+    >>> rho_random = random_density_matrix(4, seed=42)
+
+    returns a random 4x4 complex matrix like this
+
+    >>> print(rho_random)
+    [[0.23434155+0.j         0.20535572-0.04701708j 0.11523158-0.02017518j
+      0.18524981-0.13636277j]
+     [0.20535572+0.04701708j 0.2544268 +0.j         0.14478708-0.02004061j
+      0.22496295-0.11837381j]
+     [0.11523158+0.02017518j 0.14478708+0.02004061j 0.11824612+0.j
+      0.09088248-0.05551508j]
+     [0.18524981+0.13636277j 0.22496295+0.11837381j 0.09088248+0.05551508j
+      0.39298553+0.j        ]]
+
+    >>> sigma, F = operator_sinkhorn(rho=rho_random, dim=[2, 2])
+
+    This returns the result density matrix along with the operations list F.
+    SIGMA has all of its (single-party) reduced density matrices
+    proportional to the identity, and SIGMA = Tensor(F)*RHO*Tensor(F)'. In
+    other words, F contains invertible local operations that demonstrate
+    that RHO and SIGMA are locally equivalent.
+
```

Comment:
Most of these symbols should be in latex. See existing examples in toqito, if you are new to latex.